common/src/main/java/org/apache/sedona/common/raster/RasterConstructors.java

```diff
@@ -96,10 +96,8 @@ public static GridCoverage2D makeEmptyRaster(int numBand, int widthInPixel, int
         // Create a new empty raster
         WritableRaster raster = RasterFactory.createBandedRaster(DataBuffer.TYPE_DOUBLE, widthInPixel, heightInPixel, numBand, null);
         MathTransform transform = new AffineTransform2D(scaleX, skewY, skewX, -scaleY, upperLeftX + scaleX / 2, upperLeftY - scaleY / 2);
-        GridGeometry2D gridGeometry = new GridGeometry2D(new GridEnvelope2D(0, 0, widthInPixel, heightInPixel), transform, crs);
-        ReferencedEnvelope referencedEnvelope = new ReferencedEnvelope(gridGeometry.getEnvelope2D());
         // Create a new coverage
         GridCoverageFactory gridCoverageFactory = CoverageFactoryFinder.getGridCoverageFactory(null);
-        return gridCoverageFactory.create("genericCoverage", raster, referencedEnvelope);
+        return gridCoverageFactory.create("genericCoverage", raster, crs, transform, null, null, null, null, null);
```

Comment:
Since you modified the raster constructor, please check if new test cases should be added.